docs/components/List/Web.stories.tsx

```diff
@@ -20,63 +32,258 @@ const BasicTemplate: ComponentStory<typeof List> = args => (
+    align-items: center;
+    gap: var(--space-small);
+  }
+`;
+
+function RenderServiceProviderList({
+  listItem,
+}: {
+  readonly listItem: SPListItemProps;
+}) {
+  return (
+    <div className="my-item">
+      <style>{itemCSS}</style>
+      <Avatar imageUrl={listItem.avatarUrl} />
+      <Heading level={4}>{listItem.name}</Heading>
+      <div style={{ marginLeft: "auto" }}>
+        <Icon name="arrowRight" size="base" color="interactive" />
+      </div>
+    </div>
+  );
+}
+
+export const ListWithCustomRenderer = BasicTemplate.bind({});
+ListWithCustomRenderer.args = {
+  items: serviceProviderListItems,
+  customRenderItem: (item: SPListItemProps) => (
+    <RenderServiceProviderList listItem={item} />
+  ),
+};
+
+interface ProductListItemProps extends BaseListItemProps {
+  readonly name: string;
+  readonly description: string;
+  readonly image: FileUpload;
+  readonly price: string;
+  readonly section: string;
+  readonly onClick?: () => void;
+}
+
+const imageFile: FileUpload = {
+  key: "abc",
+  name: "image_of_something.png",
+  type: "image/png",
+  size: 213402324,
+  progress: 1,
+  src: () => Promise.resolve("https://picsum.photos/250"),
+};
+
+const productsList: ProductListItemProps[] = [
+  {
+    id: 1,
+    name: "Lawn Mower",
+    description: "A reliable lawn mower for home or commercial use.",
+    image: imageFile,
+    price: "$250.00",
+    section: "Products",
+    onClick: () => alert("TODO: Implement onClick"),
+  },
+  {
+    id: 2,
+    name: "Hedge Trimmer",
+    description: "Electric hedge trimmer for precise shaping.",
+    image: imageFile,
+    price: "$85.00",
+    section: "Products",
+    onClick: () => alert("TODO: Implement onClick"),
+  },
+  {
+    id: 3,
+    name: "Garden Fertilizer",
+    description: "Premium fertilizer for healthy lawn growth.",
+    image: imageFile,
+    price: "$30.00",
+    section: "Products",
+    onClick: () => alert("TODO: Implement onClick"),
+  },
+  {
+    id: 4,
+    name: "Lawn Mowing Service",
+    description: "Keep your lawn tidy with regular mowing.",
+    image: imageFile,
+    price: "$50.00",
+    section: "Services",
+    onClick: () => alert("TODO: Implement onClick"),
+  },
+  {
+    id: 5,
+    name: "Weed Control Service",
+    description: "Prevent and eliminate unwanted weeds.",
+    image: imageFile,
+    price: "$40.00",
+    section: "Services",
+    onClick: () => alert("TODO: Implement onClick"),
+  },
+  {
+    id: 6,
+    name: "Hedge Trimming Service",
+    description: "Shape your hedges to perfection.",
+    image: imageFile,
+    price: "$75.00",
+    section: "Services",
+    onClick: () => alert("TODO: Implement onClick"),
+  },
+];
+
+const sectionedCSS = `
+  .product-list-item {
+    display: flex;
+    justify-content: flex-start;
+    text-align: left;
+    padding: var(--space-small);
+  }
+
+  .product-list-item > * {
+    padding: var(--space-small);
+  }
+
+  li > ul > li:not(:last-child) .product-list-item {
```

Comment:
this is a bit awkward. we've gotta increase the specificity of what we're targeting with `li > ul >` rather than just `li:not(:last-child)` because the structure is 

```
<ul>
  <li>
    <ul>
      <li></li>
    </ul>
  </li>
  <li>
    <ul>
      <li></li>
    </ul>
  </li>
</ul>
```

so we have to say that we want to target the last item in each set of nested `ul`s

if we don't then it will only avoid adding a border bottom to the very last instance, meaning the li right above a new section **would** get a border bottom